bindings/python/multibody/geometry-data.hpp

```diff
@@ -100,6 +99,20 @@ namespace pinocchio
              bp::args("self","pair_id"),
              "Deactivate the collsion pair pair_id in geomModel.collisionPairs if it exists.")
         ;
+
+#ifdef PINOCCHIO_WITH_HPP_FCL  
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+        cl
+        .def_readonly("distanceRequest",
```

Comment:
This is not the good way to deprecate it in Python. One solution is to make it deprecated officially in the Python bindings using the right flags. Could you do it?

Reply:
How do you deprecated an attribute in Python ?

Reply:
I think we should use a wrapper in C++ to send an std::cout with a deprecated message similar to the message sent by the deprecated decorator in Python. 
The best would be to have a generic wrapper. It should not be so hard to do I think.

Reply:
I think we should use a wrapper in C++ to send an std::cout with a deprecated message similar to the message sent by the deprecated decorator in Python. 
The best would be to have a generic wrapper. It should not be so hard to do I think.

Reply:
I think we should use a wrapper in C++ to send an std::cout with a deprecated message similar to the message sent by the deprecated decorator in Python. 
The best would be to have a generic wrapper. It should not be so hard to do I think.

Reply:
After double thinking, it can also be achieved with Python properties.
```python
class Foo:
  def __init__(self):
    self._deprecated_attr = None
  @property
  def attr(self):
    # warn here
    return self._deprecated_attr
  @attr.setter
  def attr(self, value):
    # warn here
    self._deprecated_attr = value
```

Reply:
OK. Please go ahead.

Reply:
I found another solution to do it directly in C++ using `default_call_policies` arguments of `boost::python::def`. I think it will make easier the deprecation ;)

Reply:
Can you be more explicit ? I implemented something with a set of macros. I will push it and let you choose whatever you prefer.

Reply:
The idea is to overload the pre_call function of the policy and to warn a deprected warning, as done in classic Python. Of course, I can have a look to your solution.

Reply:
Yours is probably better but don't waste too much time for deprecated stuff.
I pushed it to my branch deprecation_warning. My computer is suffering for re-compiling everything. I prefer not to launch CI for nothing.

Reply:
I've made a suggestion on top of your commits which seems to be simpler and able to handle both class members and functions.